### PR TITLE
notebook execution tests

### PR DESCRIPTION
**Pull request by Aylr**
## What's here

- This PR mitigates some of the substantial risks with the codgen notebooks now critical to GE workflows.
﻿

### CODE REVIEW

**Review by ayirplm — COMMENTED** [review]
Maybe all these expectations can come from file, especially if we are demoing, "    batch.expect_table_column_count_to_equal(1)
    batch.expect_table_row_count_to_equal(1313)
    batch.expect_column_values_to_be_in_set("Sex", ["female", "male"])"

**Review by ayirplm — COMMENTED** [review]
Why only action_list_operator?

**Comment by Aylr** [comment]
> Why only action_list_operator?

I'm not sure I understand your question. Can you clarify?

File: tests/render/renderer/test_notebook_renderer.py [hunk 1]
```diff
@@ -1495,3 +1500,76 @@ def test_complex_suite(warning_suite):
...
+    ep.preprocess(nb, {"metadata": {"path": uncommitted_dir}})
+
+    # Useful to inspect executed notebook
+    with open(os.path.join(uncommitted_dir, "output.ipynb"), "w") as f:
+        nbformat.write(nb, f)
+
+    # Assertions about output
+    context = DataContext(root_dir)
+    obs_validation_result = context.get_validation_result("warning")
+    assert obs_validation_result.statistics == {
```
Comment (jcampbell) [inline tests/render/renderer/test_notebook_renderer.py:1571]: Could we assert that the new expectation suite is the same as the old expectation suite?

File: tests/render/renderer/test_notebook_renderer.py [hunk 2]
```diff
@@ -1495,3 +1500,76 @@ def test_complex_suite(warning_suite):
...
+    suite_name = "warning"
+
+    context.create_expectation_suite(suite_name)
+    csv_path = os.path.join(root_dir, "..", "data", "Titanic.csv")
+    batch_kwargs = {"datasource": "mydatasource", "path": csv_path}
+    batch = context.get_batch(batch_kwargs, suite_name)
+    batch.expect_table_column_count_to_equal(1)
+    batch.expect_table_row_count_to_equal(1313)
+    batch.expect_column_values_to_be_in_set("Sex", ["female", "male"])
+    batch.save_expectation_suite(discard_failed_expectations=False)
```
Comment (jcampbell) [inline tests/render/renderer/test_notebook_renderer.py:1533]: Could we also save the expectation suite here and the immediately delete from disk after calling the edit suite command to ensure that the notebook actually creates the new suite by being run?

**Comment by ayirplm** [comment]
> > Why only action_list_operator?
> 
> I'm not sure I understand your question. Can you clarify?

Just mean WarningAndFailureExpectationSuitesValidationOperator is missing

File: tests/render/renderer/test_notebook_renderer.py [hunk 3]
```diff
@@ -1495,3 +1500,76 @@ def test_complex_suite(warning_suite):
...
+    - add a few expectations (both table and column level)
+    - verify that no validations have happened
+    - create the suite edit notebook by hijacking the private cli method
+
+
+    We then:
+    - execute that notebook (Note this will raise various errors like
+    CellExecutionError if any cell in the notebook fails
+    - create a new context from disk
+    - a validation has been run with our expectation suite
```
Comment (jcampbell) [inline tests/render/renderer/test_notebook_renderer.py]:
```suggestion
    - verify that a validation has been run with our expectation suite
```
Small word change, and maybe a chance here to add the additional tests suggested below

**Review by jcampbell — APPROVED** [review]
A couple possible additions, but this seems like an unambiguous improvement!

File: tests/render/renderer/test_notebook_renderer.py (hunk 2 above)
Comment (Aylr) [reply tests/render/renderer/test_notebook_renderer.py:1533]: I wish we could, but the notebook itself reads the suite from disk. I thought about cracking the notebook open but it seemed like too much work and could introduce other errors.

File: tests/render/renderer/test_notebook_renderer.py (hunk 1 above)
Comment (Aylr) [reply tests/render/renderer/test_notebook_renderer.py:1571]: ooh yes nice

Comment (Aylr) [reply tests/render/renderer/test_notebook_renderer.py:1571]: Done

File: tests/render/renderer/test_notebook_renderer.py (hunk 3 above)
Comment (Aylr) [reply tests/render/renderer/test_notebook_renderer.py]: Good catch. Done

**Pull request merged**